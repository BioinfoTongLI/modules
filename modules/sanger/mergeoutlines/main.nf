process MERGEOUTLINES {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'quay.io/cellgeni/tiled_cellpose:0.1.1':
        'quay.io/cellgeni/tiled_cellpose:0.1.1' }"

    input:
    tuple val(meta), path(outlines)

    output:
    tuple val(meta), path("${prefix}_merged.wkt"), emit: multipoly_wkts
    tuple val(meta), path("${prefix}_merged.geojson"), emit: multipoly_geojsons, optional: true
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    merge_wkts.py run \\
        --sample_id ${prefix} \\
        $args \\
        $outlines

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mergeoutlines: \$(merge_wkts.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_merged.wkt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mergeoutlines: \$(merge_wkts.py version)
    END_VERSIONS
    """
}
