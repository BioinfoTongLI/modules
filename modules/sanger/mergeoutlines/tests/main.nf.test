nextflow_process {

    name "Test Process MERGEOUTLINES"
    script "../main.nf"
    config "./nextflow.config"
    process "MERGEOUTLINES"

    tag "modules"
    tag "modules_sanger"
    tag "mergeoutlines"

    test("iss - wkt") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file("/lustre/scratch126/cellgen/cellgeni/tl10/modules/modules/sanger/mergeoutlines/tests/XETG00055__0003753__Hob23-BRA-3_frontal__20230710__174614-24396_44726_28492_48822-diam_40_cp_outlines.wkt", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("iss - wkt - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file("/lustre/scratch126/cellgen/cellgeni/tl10/modules/modules/sanger/mergeoutlines/tests/XETG00055__0003753__Hob23-BRA-3_frontal__20230710__174614-24396_44726_28492_48822-diam_40_cp_outlines.wkt", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
