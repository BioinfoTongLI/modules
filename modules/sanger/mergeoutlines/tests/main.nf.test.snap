{
    "mouse heart - wkt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ],
                "multipoly_geojsons": [
                    
                ],
                "multipoly_wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T15:25:20.509025741"
    },
    "iss - wkt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ],
                "multipoly_geojsons": [
                    
                ],
                "multipoly_wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T15:24:39.339132587"
    },
    "mouse heart - wkt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,04f1e1f2018a2419b050cfb7b9db0f1c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,ae369b87c96ef64d94d12c034c3c7d09"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ],
                "multipoly_geojsons": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,ae369b87c96ef64d94d12c034c3c7d09"
                    ]
                ],
                "multipoly_wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,04f1e1f2018a2419b050cfb7b9db0f1c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T15:25:14.227476354"
    },
    "iss - wkt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,04f1e1f2018a2419b050cfb7b9db0f1c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,ae369b87c96ef64d94d12c034c3c7d09"
                    ]
                ],
                "2": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ],
                "multipoly_geojsons": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,ae369b87c96ef64d94d12c034c3c7d09"
                    ]
                ],
                "multipoly_wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,04f1e1f2018a2419b050cfb7b9db0f1c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ba98960bc9c13e98dd1a7e96af12ae44"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T15:24:34.447063742"
    }
}